golem-common/src/model/mod.rs

```diff
@@ -2851,48 +2852,87 @@ impl TryFrom<golem_api_grpc::proto::golem::worker::FileSystemNode> for Component
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Encode, Decode, Enum, Default)]
+#[derive(Debug, Clone, PartialEq, Serialize, Encode, Decode, Enum, Default)]
 #[serde(rename_all = "kebab-case")]
 #[oai(rename_all = "kebab-case")]
-pub enum WorkerBindingType {
+pub enum GatewayBindingType {
     #[default]
     Default,
     FileServer,
+    CorsPreflight,
```

Comment:
Used only in the REST API side to keep backward compatibility.
This is not a great structure because a type discriminator like this don't really distinguish the need of each binding

Reply:
Prior to this PR, this was used at the core logic side that opened up the possibility of creating unreliable logic.